packages/grafana-runtime/src/utils/queryResponse.ts

```diff
@@ -30,17 +39,21 @@ interface DataResponse {
  *
  * @public
  */
-export function toDataQueryResponse(res: any, queries?: DataQuery[]): DataQueryResponse {
+export function toDataQueryResponse(
+  res: FetchResponse<BackendDataSourceResponse | undefined>,
+  queries?: DataQuery[]
+): DataQueryResponse {
   const rsp: DataQueryResponse = { data: [], state: LoadingState.Done };
+  // If the response isn't in a correct shape we just ignore the data and pass empty DataQueryResponse.
```

Comment:
@marefr this strikes me as bit weird. Shouldn't we throw an error or something if the response isn't the correct type? Right now this is silently ignored and data dropped which doesn't seem ideal.